### PR TITLE
setup with vite

### PR DESCRIPTION
**Pull request by rutvik-balar**
make project react-TS project with vite and add eslint and prettier

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  semi: true,
+  trailingComma: 'all',
+  singleQuote: true,
+  printWidth: 90,
+  tabWidth: 2,
+  endOfLine: 'auto',
+};
+
+// {
```
Comment (r-aritra) [inline .prettierrc.js:10]: Please remove commented code.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "outdoor-event-fe",
+  "version": "0.0.0",
+  "scripts": {
+    "dev": "vite",
```
Comment (r-aritra) [inline package.json:5]: please build with tsc and then deploy with vite
"tsc & vite"

File: package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "postinstall": "husky install",
+    "build": "tsc && vite build",
+    "serve": "vite preview",
+    "lint:fix": "eslint ./src --ext .jsx,.js,.ts,.tsx --quiet --fix --ignore-path ./.gitignore",
+    "lint:format": "prettier  --loglevel warn --write \"./**/*.{js,jsx,ts,tsx,css,md,json}\" ",
+    "lint": "yarn lint:format && yarn lint:fix ",
+    "type-check": "tsc"
+  },
+  "dependencies": {
```
Comment (r-aritra) [inline package.json:14]: please install all the dependencies listed in the tech stack in the readme file

File: package.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    "@mantine/hooks": "^7.3.2",
+    "@testing-library/jest-dom": "^6.1.5",
+    "@testing-library/react": "^14.1.2",
+    "@types/jest": "^29.5.11",
+    "axios": "^1.6.2",
+    "dayjs": "^1.11.10",
+    "eslint-plugin-storybook": "^0.6.15",
+    "i18next": "^23.7.11",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^29.7.0",
```
Comment (r-aritra) [inline package.json:30]: Please remove "jest" from dependency and add it as a dev dependency
npm install jest --save-dev

File: package.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    "jest-environment-jsdom": "^29.7.0",
+    "jotai": "^2.6.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-query": "^3.39.3",
+    "react-router-dom": "^6.21.1",
+    "storybook": "^7.6.6",
```
Comment (r-aritra) [inline package.json:37]: This is also a dev dependency

File: package.json [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    "ts-jest": "^29.1.1",
```
Comment (r-aritra) [inline package.json:38]: This is also a dev dependency

File: package.json (hunk 3 above)
Comment (rutvik-balar) [reply package.json:17]: resolved

File: package.json (hunk 2 above)
Comment (rutvik-balar) [reply package.json]: resolved

File: .prettierrc.js (hunk 1 above)
Comment (rutvik-balar) [reply .prettierrc.js]: resolved

File: package.json (hunk 4 above)
Comment (rutvik-balar) [reply package.json:30]: resolved

File: package.json (hunk 5 above)
Comment (rutvik-balar) [reply package.json:37]: resolved

File: package.json (hunk 6 above)
Comment (rutvik-balar) [reply package.json]: resolved

**Review by r-aritra — APPROVED** [review]
lgtm

**Pull request merged**